ssl/ssl_lib.c

```diff
@@ -4360,6 +4361,72 @@ const CTLOG_STORE *SSL_CTX_get0_ctlog_store(const SSL_CTX *ctx)
     return ctx->ctlog_store;
 }
 
+void SSL_CTX_set_early_cb(SSL_CTX *c, SSL_early_cb_fn cb, void *arg)
+{
+    c->early_cb = cb;
+    c->early_cb_arg = arg;
+}
+
+int SSL_early_isv2(SSL *s)
+{
+    return s->clienthello->isv2;
+}
+
+unsigned int SSL_early_get0_legacy_version(SSL *s)
+{
+    return s->clienthello->legacy_version;
```

Comment:
Generally our pattern seems to be we don't check the primary object (i.e. SSL) for NULL, but we do check other things. It would only be NULL through programmer error (calling when not in an early callback), but - still, we probably shouldn't crash.